### PR TITLE
boards: pinetime_devkit0: add compatible for generic spi nor flash

### PR DESCRIPTION
**Pull request by ck-telecom**
using the correct datasheet download from official site: http://www.xtxtech.com/download/?AId=154

Signed-off-by: Qingsong Gou <gouqs@hotmail.com>

### CODE REVIEW

File: boards/arm/pinetime_devkit0/pinetime_devkit0.dts [hunk 1]
```diff
@@ -124,12 +124,13 @@
...
-	mx25l: mx25l@0 {
-		compatible = "mxicy,cmos-mx25l";
+	xt25fb32: xt25fb32@0 {
+		compatible = "jedec,spi-nor";
 		reg = <0>;
 		spi-max-frequency = <8000000>; /* 8MHz */
-		label = "CMOS MX25L";
+		label = "XT25FB32";
+		jedec-id = [0b 40 16];
+		size = <0x2000000>;
```
Comment (MaureenHelm) [inline boards/arm/pinetime_devkit0/pinetime_devkit0.dts:133]: Recommend using DT_SIZE_M from `dts/common/mem.h`

File: boards/arm/pinetime_devkit0/pinetime_devkit0.dts [hunk 2]
```diff
@@ -124,12 +124,13 @@
...
-	mx25l: mx25l@0 {
-		compatible = "mxicy,cmos-mx25l";
+	xt25fb32: xt25fb32@0 {
+		compatible = "jedec,spi-nor";
 		reg = <0>;
 		spi-max-frequency = <8000000>; /* 8MHz */
-		label = "CMOS MX25L";
+		label = "XT25FB32";
+		jedec-id = [0b 40 16];
+		size = DT_SIZE_M(32);
```
Comment (MaureenHelm) [inline boards/arm/pinetime_devkit0/pinetime_devkit0.dts:133]:
```suggestion
		size = <DT_SIZE_M(32)>;
```

**Review by MaureenHelm — DISMISSED** [review]

**Comment by ck-telecom** [comment]
It seems that CI error depends on #41325

**Review by MaureenHelm — APPROVED** [review]

**Comment by ck-telecom** [comment]
@gmarull could you please approve this PR?

**Review by gmarull — APPROVED** [review]

**Pull request merged**